pkg/config/adapterInfoRegistry.go

```diff
@@ -27,47 +28,104 @@ type adapterInfoRegistry struct {
 	adapterInfosByName map[string]*adapter.BuilderInfo
 }
 
+// DNS1035Str defines regex to test for DNS-1035 compatible names.
+const DNS1035Str = "^[a-z]([-a-z0-9]*[a-z0-9])?$"
+
+// DNS1035 is the prebuilt matcher for DNS-1035 compatible names.
+var DNS1035 = regexp.MustCompile(DNS1035Str)
+
+func dns1035Err(name string) error {
+	return fmt.Errorf("%s cannot be converted to a DNS-1035 compatible name %s", name, DNS1035Str)
+}
+
+// NameToCfgKind returns Kind name given an adapter name.
+// istio.io/mixer/adapter/denier --> denier-mixer-istio-io
+// github.com/user1/mixerAdapters/denier --> denier-mixeradapters-user1
+// metricsco.io/mixerAdapters/metrics1 --> metrics1-mixeradapters-metricso-io
+// metricsco.io/mixerAdapters/adapters/metrics2 --> metrics2-mixeradapters-metricsco-io
+// kind = $pkg-$reponame-$org
+// Resulting name is DNS-1035 compatible. [a-z]([-a-z0-9]*[a-z0-9])?
+func NameToCfgKind(adapterName string) (string, error) {
```

Comment:
I don't like the way this works.

I don't think we should treat github specially, we shouldn't have any special cases here. Just take the incoming slash-based name, flip it and dashify it.